package NYT.NChunkClient.NProto;

import "yt/ytlib/node_tracker_client/node.proto";
import "yt/ytlib/chunk_client/chunk.proto";
import "yt/ytlib/object_client/master_ypath.proto";

////////////////////////////////////////////////////////////////////////////////

message TReqConfirm
{
    repeated fixed32 replicas = 1;
    required NChunkClient.NProto.TChunkMeta chunk_meta = 2;
    required NChunkClient.NProto.TChunkInfo chunk_info = 3;
}

message TRspConfirm
{ }

////////////////////////////////////////////////////////////////////////////////

// CreateObjects extension.

message TReqCreateChunkExt
{
    extend NObjectClient.NProto.TReqCreateObjects
    {
        optional TReqCreateChunkExt create_chunk_ext = 100;
    }

    // Requested number of nodes to write.
    optional int32 upload_replication_factor = 1 [default = 1];
 
    // Desired number of copies (including those created by background replication).
    optional int32 replication_factor = 2 [default = 1];
 
    // Host name (without port number!) to be preferred for upload.
    // The client typically puts the name of the local host here to
    // facilitate local uploads.
    optional string preferred_host_name = 3;

    // See NErasure::ECodec.
    optional int32 erasure_codec = 4 [default = 0];

    // Can this chunk be balanced to other nodes?
    optional bool movable = 5 [default = true];

    required bool vital = 6 [default = false];
}

message TRspCreateChunkExt
{
    extend NObjectClient.NProto.TRspCreateObjects
    {
        optional TRspCreateChunkExt create_chunk_ext = 100;
    }

    required NNodeTrackerClient.NProto.TNodeDirectory node_directory = 1;
    repeated fixed32 replicas = 2;
}

////////////////////////////////////////////////////////////////////////////////
